google/cloud/bigquery/schema.py

```diff
@@ -291,3 +318,72 @@ def _to_schema_fields(schema):
         field if isinstance(field, SchemaField) else SchemaField.from_api_repr(field)
         for field in schema
     ]
+
+class PolicyTagList(object):
+    """Define Policy Tags for a column.
+
+    Args:
+        names (Union[List[str], None]): list of policy tags to associate with
+            the column.
+    """
+
+    def __init__(self, names=None):
+        self._properties = {}
+        if names is not None:
+            self.names = names
+
+    @property
+    def names(self):
+        """Union[List[str], None]: Policy tags associated with this definition.
+        """
+        return self._properties.get("names", ())
+
+    @names.setter
+    def names(self, value):
+        """Union[List[str], None]: Policy tags associated with this definition.
+
+        (Defaults to :data:`None`).
+        """
+        if value is not None:
+            self._properties["names"] = value
+        else:
+            if "names" in self._properties:
+                del self._properties["names"]
+
+    @classmethod
+    def from_api_repr(cls, api_repr):
+        """Return a :class:`PolicyTagList` object deserialized from a dict.
+
+        This method creates a new ``PolicyTagList`` instance that points to
+        the ``api_repr`` parameter as its internal properties dict. This means
+        that when a ``PolicyTagList`` instance is stored as a property of
+        another object, any changes made at the higher level will also appear
+        here.
+
+        Args:
+            api_repr (Mapping[str, str]):
+                The serialized representation of the PolicyTagList, such as
+                what is output by :meth:`to_api_repr`.
+
+        Returns:
+            google.cloud.bigquery.schema.PolicyTagList:
+                The ``PolicyTagList`` object.
+        """
+        instance = cls()
+        instance._properties = api_repr
```

Comment:
Let's perhaps assign a copy of the input arg instead? If some code later modifies the input dict, the instance itself will also be affected, which is probably not a desired side effect.

Reply:
copy.deepcopy it is